pkg/ast/pipesearch/searchHandler.go

```diff
@@ -479,10 +497,30 @@ func convertRRCsToJSONResponse(rrcs []*sutils.RecordResultContainer, sizeLimit u
 		return allJsons, allCols, err
 	}
 
+	// Filter out null values from each record
+	filteredJsons := make([]map[string]interface{}, len(allJsons))
+	for i, record := range allJsons {
```

Comment:
creating a copy of every record and removing the null values from the copy is going to be expensive (memory and cpu wise). 
we should just modify the allJsons map, removing things that are null